### ISSUE TITLE
It would be nice if we supported mcelog and/or edac-util

### ISSUE THREAD

**Issue by dswartz** [astersmith/esos#137]
Running server grade HW to serve up block storage.  Being able to find out if ECC errors are occurring or whatever, would be nice.  I will look at what is involved in doing this.

**Comment by msmith626** [comment]
Yes, those sound like good additions. If you could post links to tarballs of each (latest and greatest), I'll get them added to master.

--Marc

**Comment by dswartz** [comment]
On 2015-07-09 15:18, Marc A. Smith wrote:
> Yes, those sound like good additions. If you could post links to
> tarballs of each (latest and greatest), I'll get them added to master.

10-4

**Comment by dswartz** [comment]
mcelog seems to only provide the SW via git:

git://git.kernel.org/pub/scm/utils/cpu/mce/mcelog.git 

For edac:

http://downloads.sourceforge.net/project/edac-utils/edac-utils/0.16/edac-utils-0.16.tar.bz2

we may not need edac.  the FAQ here seems to indicate it's redundant to mcelog on modern boxes:

http://mcelog.org/faq.html

**Comment by msmith626** [comment]
Thanks; from the FAQ:
--snip--
Can you release mcelog?

mcelog is on a rolling release through the git tree. However if you need a
specific version in the git tree, and a git sha identifier is not good
enough, you can use the "vXXX" tags which are regularly incremented.
--snip--

So I'll just pull a copy from the latest tag; I read the FAQ entry about
EDAC and I'll include both in ESOS as some users may have AMD hardware and
I have to imagine it ends up being a pretty small binary. Should have it
done later this week.


--Marc

**Comment by msmith626** [comment]
Sorry for the delay... I just pushed commits to master that include the mcelog and edac-utils packages... I haven't tested the tools myself yet, so feedback would be appreciated. Also, if you'd be interested, what do you think about adding these tools to the health_chk.sh script?

--Marc

**Comment by dswartz** [comment]
On 2015-08-01 14:37, Marc A. Smith wrote:
> Sorry for the delay... I just pushed commits to master that include
> the mcelog and edac-utils packages... I haven't tested the tools
> myself yet, so feedback would be appreciated. Also, if you'd be
> interested, what do you think about adding these tools to the
> health_chk.sh script?

Probably a good idea!  I will do a build from master later today and 
give it a shot...

**Comment by dswartz** [comment]
I didn't have a chance to actually install the entire build on my SAN, but I did copy over mcelog executable and am running it in daemon mode.  Nothing has been logged yet, as I have no idea how to simulate a MCE.

**Comment by dswartz** [comment]
Although I did find this:
http://stackoverflow.com/questions/7501886/how-to-simulate-or-trigger-a-machine-check-exception

**Comment by msmith626** [comment]
Thanks; I haven't forgotten about this -- I'm not sure I will include that tool in the ESOS build, but I will compile it and do some testing to work on the health_chk.sh script...

--Marc

**Comment by msmith626** [comment]
I've decided not to take any additional action on this... both of the binaries (and related tools) are included in the image (master branch). I tested a bit and these tools are really dependent on the hardware they run on, the servers... I don't want to put generic tests in when they are known to produce error messages are not work if the hardware simply isn't supported. So, maybe in the future we can make some more exotic tests, but nothing for now.

--Marc
